addons/purchase_stock/models/stock_rule.py

```diff
@@ -115,11 +116,11 @@ def _run_buy(self, procurements):
 
     @api.model
     def _get_procurements_to_merge_groupby(self, procurement):
-        return procurement.product_id, procurement.product_uom
+        return procurement.product_id, procurement.product_uom, procurement.values.get('rule').id
```

Comment:
You would not allow to have two product merged with the same expected date just because they come from 2 different rules. It's rather technical than functional the user probably want to see only 1 line, he would not understand the split.

Reply:
We have to split by rule because it might be possible that rule has a different delay ... 
and if rule is same then it will be going to merge.

Reply:
and if 2rules have the same delay? In this case you should use the date_expected + the rule_delay in order to know if you merge it or not

Reply:
Your case is correct but it is increasing no of fields.

 1) What if we have different rules ( currently i have split it because we can use it for different picking type ) 
2) We have added directly rule_id ( propagate_date, propagate_date_minimum_delta ) to decrease fields and data on the move model.

`abs(delta_days) >= move.rule_id.propagate_date_minimum_delta`

if you want to merge those move with the same propagate_date_minimum_delta and propagate_date ( # We need to remove rule_id on move, POL ) and add those two fields

So suggest us which way you prefer.

Reply:
Currently, I prefer to add both field propagate_date_minimum_delta and propagate_date on stock_move.